### PR TITLE
[rethinkdb] Adds missing method definitions and formats the file

### PR DESCRIPTION
**Pull request by alexgorbatchev**
Please see 5ac67c6 and 317be6bc69da5f69b9517f1ff4c7908745dd5592 for actual changes, not including the formatting.

cc @seanhess @Igorbek @mhegazy @vvakame 

Also fixes the following issues:

```
rethinkdb/rethinkdb.d.ts(38,5): error TS7010: 'each', which lacks return-type annotation, implicitly has an 'any' return type.
rethinkdb/rethinkdb.d.ts(39,5): error TS7010: 'each', which lacks return-type annotation, implicitly has an 'any' return type.
rethinkdb/rethinkdb.d.ts(40,5): error TS7010: 'next', which lacks return-type annotation, implicitly has an 'any' return type.
rethinkdb/rethinkdb.d.ts(41,5): error TS7010: 'toArray', which lacks return-type annotation, implicitly has an 'any' return type.
rethinkdb/rethinkdb.d.ts(42,5): error TS7010: 'close', which lacks return-type annotation, implicitly has an 'any' return type.
rethinkdb/rethinkdb.d.ts(53,5): error TS7010: 'close', which lacks return-type annotation, implicitly has an 'any' return type.
rethinkdb/rethinkdb.d.ts(55,5): error TS7010: 'use', which lacks return-type annotation, implicitly has an 'any' return type.
rethinkdb/rethinkdb.d.ts(56,5): error TS7010: 'addListener', which lacks return-type annotation, implicitly has an 'any' return type.
rethinkdb/rethinkdb.d.ts(57,5): error TS7010: 'on', which lacks return-type annotation, implicitly has an 'any' return type.
```

### CODE REVIEW

**Review by Igorbek — APPROVED** [review]

**Comment by alexgorbatchev** [comment]
@vvakame how do I get this merged? Doesn't seem like the @dt-bot picked up on this PR for some reason...

File: rethinkdb/rethinkdb.d.ts [hunk 1]
```diff
@@ -36,10 +36,15 @@ declare module "rethinkdb" {
     export class Cursor {
         hasNext(): boolean;
         each(cb: (err: Error, row: any) => void, done?: () => void): void;
+        each<T>(cb: (err: Error, row: T) => void, done?: () => void): void;
```
Comment (vladima) [inline rethinkdb/rethinkdb.d.ts]: this `T` cannot be inferred from anything unless user has provided it via type annotation on parameter which means that user has to either write type argument explicitly `each<Foo>((err, v) => ...)` or specify type annotation on the parameter: `each((err, v: Foo) => ...)`. Just curious is this the intended behavior?

Comment (alexgorbatchev) [inline rethinkdb/rethinkdb.d.ts]: @vladima I'm not completely clear about your question.

My intent is to allow support for typed `each` method in addition to untyped, eg:

```
cursor.each<IMyRowType>((err: Error, row: IMyRowType) => ...);
```

Is this a wrong way to approach this?

Comment (vladima) [inline rethinkdb/rethinkdb.d.ts]: no, it is fine, just want to be sure that this is the intention

**Comment by alexgorbatchev** [comment]
@vladima  Thank you for accepting the PR! What does `Revision requested` label mean? Should I change something?

**Pull request merged**